react-graphql-apollo-tutorial.md

```diff
@@ -1405,29 +1401,27 @@ const updateAddStar = (
-
-After all, it is a best practice to have the identical fragment for all three parts: the initial query, the `readFragment()` and `writeFragment()` cache method. It is because your data structure for the particular entity stays consistent in your cache and there a not various definitions of this entity. For instance, if you forget to include a property which is defined by the fragment's fields in data object of the `writeFragment()` method, you will get a warning: *Missing field __typename in ...*. All of this helps you to embrace best practices when using Apollo Client.
+Let's recap all three steps here. First, you have retrieved (read) the repository entity from the Apollo Client using an identifier and the fragment; second, you updated the information of the entity; and third, you wrote back the data with updated information, but kept all remaining information intac using the JavaScript spread operator. This is a manual update mechanism that can be used when a mutation is missing data.
 
-On a implementation detail level, you have learned about extracting fragments from your query (or mutation). Basically fragments allow you to define your shared entities by GraphQL types. Afterward, you can reuse those in your queries, mutations or local state management methods to update the cache.
+It is a good practice to use an identical fragment for all three parts: the initial query, the `readFragment()`, and `writeFragment()` cache method. Your data structure for the entity stays consistent in your cache. For instance, if you forget to include a property defined by the fragment's fields in data object of the `writeFragment()` method, you  get a warning: *Missing field __typename in ...*. 
 
-On a higher level, you have learned that Apollo Client's cache normalizes your data. Only this way you are able to retrieve entities that were fetched with a deeply nested query, by using their type and identifier as composite key. Otherwise, imagine you would have to perform all the normalization of the fetched data yourself before putting it in your store/state.
+On an implementation level, you learned about extracting fragments from a query (or mutation. Fragments allow you to define your shared entities by GraphQL types.  You can reuse those in your queries, mutations or local state management methods to update the cache. On a higher level, you learned that Apollo Client's cache normalizes your data, so you can retrieve entities that were fetched with a deeply nested query using their type and identifier as composite key. Without it, you'd have to perform normalizations for all the fetched data before putting it in your store/state.
```

Comment:
"On an implementation level, you learned about extracting fragments from a query (or mutation." to "On an implementation level, you learned about extracting fragments from a query (or mutation)."

Reply:
Remove: "Fragments allow you to define your shared entities by GraphQL types."

Reply:
Parenthesis are usually unnecessary.